src/zenml/integrations/skypilot/orchestrators/skypilot_gcp_vm_orchestrator.py

```diff
@@ -0,0 +1,73 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the a Skypilot based GCP VM orchestrator."""
+
+from typing import TYPE_CHECKING, Optional, Type, cast
+
+import sky
+
+from zenml.integrations.gcp.google_credentials_mixin import (
+    GoogleCredentialsMixin,
+)
+from zenml.integrations.skypilot.flavors.skypilot_orchestrator_gcp_vm_flavor import (
+    SkypilotGCPOrchestratorConfig,
+    SkypilotGCPOrchestratorSettings,
+)
+from zenml.integrations.skypilot.orchestrators.skypilot_base_vm_orchestrator import (
+    SkypilotBaseOrchestrator,
+)
+from zenml.logger import get_logger
+
+if TYPE_CHECKING:
+    from zenml.config.base_settings import BaseSettings
+
+logger = get_logger(__name__)
+
+
+class SkypilotGCPOrchestrator(
+    SkypilotBaseOrchestrator, GoogleCredentialsMixin
+):
+    """Orchestrator responsible for running pipelines remotely in a VM.
+
+    This orchestrator does not support running on a schedule.
+    """
+
+    DEFAULT_INSTANCE_TYPE: str = "n1-standard-4"
```

Comment:
For my curiosity, why is this the default? Seems like already a not-insignificant machine to spin up. https://cloud.google.com/compute/docs/general-purpose-machines#n1_machines

Reply:
Maybe this is just the default recommended by skypilot?

Reply:
that's right, it's the same one given by skypilot. Do you think we should default to some more powerful one?

Reply:
No it's ok. Just was curious why we chose that one. We can leave it as it is.